### PR TITLE
feat: adjust Tabs to newest 0.5.0 fundamental-styles structure

### PR DESCRIPTION
**Pull request by JKMarkowski**
#### Please provide a link to the associated issue.
fixes: https://github.com/SAP/fundamental-ngx/issues/1844
#### Please provide a brief summary of this pull request.
0.5.0 styles tabs adaptation, it came with a lot of new tab modes
`Added 3 new modes for tab component process, filter, icon-only, which can be enabled by passing [mode]="name of mode into fd-tab-list or fd-tab-nav.`
Also there is new compact mode.
#### Please check whether the PR fulfills the following requirements

- [x] the commit message follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/master/CONTRIBUTING.md
- [x] tests for the changes that have been done

Documentation checklist:
- [x] update `README.md`
- [x] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)
- [x] Documentation Examples
- [x] Stackblitz works for all examples

### CODE REVIEW

File: libs/core/src/lib/tabs/tab-list.component.ts [hunk 1]
```diff
@@ -112,18 +134,14 @@ export class TabListComponent implements AfterContentInit, OnChanges, OnDestroy
...
+    private _isIndexInRange(): boolean {
         return this.panelTabs && this.panelTabs.length > 0 && this.selectedIndex < this.panelTabs.length;
     }
 
-    private isTargetTabEnabled(index: number): boolean {
-        return !this.panelTabs.toArray()[index].disabled;
-    }
-
-    private isTabContentEmpty(): boolean {
+    private _isTabContentEmpty(): boolean {
```
Comment (salarenko) [inline libs/core/src/lib/tabs/tab-list.component.ts]: Can be simplified to:
```
    private _isTabContentEmpty(): boolean {
        return !this.panelTabs.some(tab => tab.expanded)
    }
```
Also I think method name is misleading it suggests checking if existing tab content has anything inside but in fact it checks if any tab is currently opened.

File: libs/core/src/lib/tabs/tab-list.component.ts [hunk 2]
```diff
@@ -133,7 +151,7 @@ export class TabListComponent implements AfterContentInit, OnChanges, OnDestroy
         return result;
     }
 
-    private resetTabHook(): void {
+    private _resetTabHook(): void {
         this.selectedIndex = 0;
```
Comment (salarenko) [inline libs/core/src/lib/tabs/tab-list.component.ts]: This `this.selectedIndex = 0;` is doubled in `selectTab` method.

File: libs/core/src/lib/tabs/tab-nav/tab-nav.component.ts [hunk 3]
```diff
@@ -1,9 +1,9 @@
 import {
     AfterContentInit, ChangeDetectionStrategy,
     Component,
-    ContentChildren,
-    EventEmitter,
-    OnDestroy,
+    ContentChildren, ElementRef,
+    EventEmitter, Input,
+    OnDestroy, OnInit,
```
Comment (salarenko) [inline libs/core/src/lib/tabs/tab-nav/tab-nav.component.ts]: Unused import

File: libs/core/src/lib/tabs/tab-nav/tab-nav.component.ts [hunk 4]
```diff
@@ -91,7 +98,9 @@ export class TabNavComponent implements AfterContentInit, OnDestroy {
             this.renderer.listen(linkElement.elementRef.nativeElement, 'keydown', (event) => {
```
Comment (salarenko) [inline libs/core/src/lib/tabs/tab-nav/tab-nav.component.ts]: Can we move this to separate methods?

File: libs/core/src/lib/tabs/tabs.module.ts [hunk 5]
```diff
@@ -29,7 +45,16 @@ import { TabItemDirective } from './tab-item/tab-item.directive';
...
+        TabLinkDirective,
+        TabTagDirective,
+        TabIconDirective,
+        TabCountDirective,
+        TabLabelDirective,
+        TabProcessDirective,
+        TabHeaderDirective,
+        TabCounterHeaderDirective,
+        TabProcessIconDirective,
+        TabSeparator
```
Comment (InnaAtanasova) [inline libs/core/src/lib/tabs/tabs.module.ts]: These should also be exported in the `api-files.ts` fille

File: apps/docs/src/app/core/component-docs/tabs/examples/tabs-example.component.html [hunk 6]
```diff
@@ -1,14 +1,94 @@
...
+
+<label fd-form-label>Simple Tabs Built by Defining Template</label>
+<fd-tab-list>
+    <fd-tab>
+        <ng-template fd-tab-title-template>
+            <span fd-tab-tag>Link 1</span>
+        </ng-template>
+        Content for tab 1
+    </fd-tab>
+    <fd-tab [title]="'Link 2'">
```
[the remark this answers is not in the document]
Comment (JKMarkowski) [reply apps/docs/src/app/core/component-docs/tabs/examples/tabs-example.component.html]: Title Removed

File: apps/docs/src/app/core/component-docs/tabs/examples/tab-process-example/tab-process-example.component.html [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+
+<br/>
+<br/>
+
+<label fd-form-label>Process Tabs Built by defining own template with routerLink</label>
+<nav fd-tab-nav [mode]="'process'">
+    <div fd-tab-item>
+        <a fd-tab-link
+           [active]="rla1.isActive"
+           [routerLink]="['tab1', {outlets: {aux: 'process'}}]"
```
[the remark this answers is not in the document]
Comment (JKMarkowski) [reply apps/docs/src/app/core/component-docs/tabs/examples/tab-process-example/tab-process-example.component.html]: Remvoed

File: libs/core/src/lib/tabs/tab-list.component.ts (hunk 1 above)
Comment (JKMarkowski) [reply libs/core/src/lib/tabs/tab-list.component.ts]: That's true, it's leftover of old code

File: libs/core/src/lib/tabs/tab-list.component.ts [hunk 8]
```diff
@@ -94,12 +125,15 @@ export class TabListComponent implements AfterContentInit, OnChanges, OnDestroy
...
-            this.selectedIndexChange.emit(tabIndex);
+        if (this._isIndexInRange(tabIndex)) {
+            setTimeout(() => {
+                this.panelTabs.forEach((tab, index) => {
+                    tab.triggerExpandedPanel(index === tabIndex);
+                });
+                this.selectedIndex = tabIndex;
+                this.selectedIndexChange.emit(tabIndex);
+                this._changeRef.detectChanges();
+            })
```
Comment (JKMarkowski) [inline libs/core/src/lib/tabs/tab-list.component.ts]: There is no way to avoid setTimeout

File: libs/core/src/lib/tabs/tabs.module.ts (hunk 5 above)
Comment (JKMarkowski) [reply libs/core/src/lib/tabs/tabs.module.ts]: There are exported all directives from `tab-directives` in public_api.ts

File: libs/core/src/lib/tabs/tab-nav/tab-nav.component.ts (hunk 4 above)
Comment (JKMarkowski) [reply libs/core/src/lib/tabs/tab-nav/tab-nav.component.ts]: Done

File: libs/core/src/lib/tabs/tab-nav/tab-nav.component.ts (hunk 3 above)
Comment (JKMarkowski) [reply libs/core/src/lib/tabs/tab-nav/tab-nav.component.ts]: Removed

File: libs/core/src/lib/tabs/tab-list.component.ts (hunk 2 above)
Comment (JKMarkowski) [reply libs/core/src/lib/tabs/tab-list.component.ts]: Done

File: libs/core/src/lib/tabs/tab-list.component.ts [hunk 9]
```diff
@@ -52,24 +71,26 @@ export class TabListComponent implements AfterContentInit, OnChanges, OnDestroy
...
     constructor(
-        private tabsService: TabsService
-    ) {}
+        private _tabsService: TabsService,
+        private _changeRef: ChangeDetectorRef
+    ) {
+    }
 
     /** @hidden */
     ngAfterContentInit(): void {
```
[the remark this answers is not in the document]
Comment (JKMarkowski) [reply libs/core/src/lib/tabs/tab-list.component.ts]: Done

File: libs/core/src/lib/tabs/tab-list.component.ts [hunk 10]
```diff
@@ -40,52 +46,77 @@ export class TabListComponent implements AfterContentInit, OnChanges, OnDestroy
...
+
+        this._tabSelectSubscription = this._tabsService.tabSelected
+            .pipe(
+                takeUntil(this.onDestroy$),
+                filter(index => index !== this.selectedIndex)
+            )
+            .subscribe(index => this.selectTab(index))
+        ;
+
+        this._tabsSubscription = this.panelTabs.changes
```
Comment (JKMarkowski) [inline libs/core/src/lib/tabs/tab-list.component.ts]: Remove subscriptions reference

File: libs/core/src/lib/tabs/tab-list.component.ts (hunk 9 above)
[the remark this answers is not in the document]
Comment (salarenko) [reply libs/core/src/lib/tabs/tab-list.component.ts]: Not true :(

File: libs/core/src/lib/tabs/tab-nav/tab-nav.component.ts [hunk 11]
```diff
@@ -12,91 +12,121 @@ import {
...
     /** Event Thrown every time something is clicked */
-    @Output() onKeyDown = new EventEmitter<{event: any, index: number}>();
+    @Output() onKeyDown = new EventEmitter<{ event: any, index: number }>();
 
     /** @hidden */
     constructor(
         private renderer: Renderer2,
-        private tabsService: TabsService
-    ) {}
+        private tabsService: TabsService,
```
Comment (JKMarkowski) [inline libs/core/src/lib/tabs/tab-nav/tab-nav.component.ts]: `_` prefix

File: libs/core/src/lib/tabs/tab-utils/tab-directives.ts [hunk 12]
```diff
@@ -42,3 +42,154 @@ export class TabLoadTitleDirective implements OnInit {
...
+
+
+@Directive({
+    // tslint:disable-next-line:directive-selector
+    selector: '[fd-tab-separator]',
+    host: {
+        'class': 'fd-tabs__separator'
+    }
+})
+export class TabSeparator {
```
Comment (JKMarkowski) [inline libs/core/src/lib/tabs/tab-utils/tab-directives.ts]: Add Directive

File: libs/core/src/lib/tabs/tab-nav/tab-nav.component.ts (hunk 4 above)
Comment (salarenko) [reply libs/core/src/lib/tabs/tab-nav/tab-nav.component.ts]: Not true :(

**Comment by JKMarkowski** [comment]
> Are there specs for the counters? The font-size on the normal counters is the same size as the tab text itself, whereas the counter font-size is smaller on the icon-only tabs which I think looks better

It's defined in documentation

File: libs/core/src/lib/tabs/tab-nav/tab-nav.component.ts (hunk 4 above)
Comment (JKMarkowski) [reply libs/core/src/lib/tabs/tab-nav/tab-nav.component.ts]: Now it is :D

**Pull request merged**